### PR TITLE
Widget collection

### PR DESCRIPTION
**Pull request by sofiaIvarsRitense**
https://ritense.tpondemand.com/entity/109489-fe-cw-widget-collection-case-tab

### CODE REVIEW

File: projects/valtimo/dossier/src/lib/components/dossier-detail/tab/widgets/components/collection/widget-collection.component.html [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    widgetConfiguration: widgetConfiguration$ | async,
+    widgetPropertyValue: widgetPropertyValue$ | async
+  } as obs"
+>
+  <span class="widget-collection__title">Main title</span>
+
+  <div
+    #widgetCollection
+    [ngClass]="{
+      'widget-collection__render-vertically': renderVertically() === 1,
```
Comment (mbritense) [inline projects/valtimo/dossier/src/lib/components/dossier-detail/tab/widgets/components/collection/widget-collection.component.html]: change renderVertically to amountOfColumns

File: projects/valtimo/dossier/src/lib/components/dossier-detail/tab/widgets/components/collection/widget-collection.component.scss [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  &__card-input {
+    background: transparent;
+  }
+
+  &__full-width-input {
+    width: 100%;
+  }
+
+  &__half-width-input {
+    width: 45%;
```
Comment (mbritense) [inline projects/valtimo/dossier/src/lib/components/dossier-detail/tab/widgets/components/collection/widget-collection.component.scss]: More precise, use calc(50% - **px)

File: projects/valtimo/dossier/src/lib/components/dossier-detail/tab/widgets/components/collection/widget-collection.component.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+          );
+        })
+      )
+    );
+
+  private _observer!: ResizeObserver;
+  private readonly _queryParams$ = new BehaviorSubject<string | null>(null);
+
+  constructor(
+    private viewContentService: ViewContentService,
```
Comment (mbritense) [inline projects/valtimo/dossier/src/lib/components/dossier-detail/tab/widgets/components/collection/widget-collection.component.ts]: private readonly

File: projects/valtimo/dossier/src/lib/components/dossier-detail/tab/widgets/components/collection/widget-collection.component.html [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      'widget-collection__render-two-columns': renderVertically() === 2,
+      'widget-collection__render-three-columns': renderVertically() === 3,
+      'widget-collection__container': renderVertically() === 4
+    }"
+  >
+    <div class="widget-collection__card" *ngFor="let field of obs?.widgetPropertyValue">
```
Comment (teodor-ritense) [inline projects/valtimo/dossier/src/lib/components/dossier-detail/tab/widgets/components/collection/widget-collection.component.html]: use @for from contorl-flow

File: projects/valtimo/dossier/src/lib/components/dossier-detail/tab/widgets/components/collection/widget-collection.component.html [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      <div
+        *ngFor="let property of field"
```
Comment (teodor-ritense) [inline projects/valtimo/dossier/src/lib/components/dossier-detail/tab/widgets/components/collection/widget-collection.component.html]: Same as above

File: projects/valtimo/dossier/src/lib/components/dossier-detail/tab/widgets/components/collection/widget-collection.component.scss [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+.widget-collection__render {
+  &-vertically,
+  &-two-columns,
+  &-three-columns {
+    display: grid;
+    grid-gap: 16px;
+  }
+
+  &-vertically {
```
Comment (teodor-ritense) [inline projects/valtimo/dossier/src/lib/components/dossier-detail/tab/widgets/components/collection/widget-collection.component.scss]: This should not be necessary...by default grid has 1fr on columns

File: projects/valtimo/dossier/src/lib/components/dossier-detail/tab/widgets/components/collection/widget-collection.component.ts [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+    InputModule,
+    PaginationModule,
+    TilesModule,
+    WidgetCollectionComponent,
+    CarbonListModule,
+    TranslateModule,
+  ],
+})
+export class WidgetCollectionComponent implements AfterViewInit, OnDestroy {
+  @HostBinding('class') public readonly class = 'widget-collection';
```
Comment (teodor-ritense) [inline projects/valtimo/dossier/src/lib/components/dossier-detail/tab/widgets/components/collection/widget-collection.component.ts]:
```suggestion
  @HostBinding('class') public readonly class = 'valtimo-widget-collection';
```

File: projects/valtimo/dossier/src/lib/components/dossier-detail/tab/widgets/components/collection/widget-collection.component.ts [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+  @ViewChild('widgetCollection') private _widgetCollectionRef: ElementRef<HTMLDivElement>;
+
+  @Input() collapseVertically = false;
+  @Input() public set widgetConfiguration(value: CollectionCaseWidget) {
+    if (!value) return;
+    this.widgetConfiguration$.next(value);
+  }
+  public readonly showPagination$ = new BehaviorSubject<boolean>(false);
+
+  @Input() public set widgetData(value: any | null) {
```
Comment (teodor-ritense) [inline projects/valtimo/dossier/src/lib/components/dossier-detail/tab/widgets/components/collection/widget-collection.component.ts]:
```suggestion
  @Input() public set widgetData(value: Page<CarbonListItem> | null) {
```

File: projects/valtimo/dossier/src/lib/components/dossier-detail/tab/widgets/components/collection/widget-collection.component.ts [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+  public readonly paginationModel = signal<PaginationModel>(new PaginationModel());
+  public readonly amountOfColumns = signal(0);
+  private widgetData$ = new BehaviorSubject<CarbonListItem[] | null>(null);
+
+  public readonly widgetPropertyValue$: Observable<{title: string; value: string}[][]> =
+    combineLatest([this.widgetConfiguration$, this.widgetData$]).pipe(
+      filter(([widget, widgetData]) => !!widget && !!widgetData),
+      map(([widget, widgetData]) =>
+        widgetData.map(widgetFieldData => {
+          return widget?.properties.fields.reduce(
```
Comment (teodor-ritense) [inline projects/valtimo/dossier/src/lib/components/dossier-detail/tab/widgets/components/collection/widget-collection.component.ts]: Single line return, no need for {}

**Review by teodor-ritense — APPROVED** [review]

**Pull request merged**